plugins/ftp/vsftpd

```diff
@@ -34,18 +62,24 @@ ftp_su=U
 ftp_fu=U
 ftp_sd=U
 ftp_fd=U
+ftp_sr=U
+ftp_fr=U
 ftp_sde=U
 ftp_fde=U
 
-ftp_c=`grep "CONNECT" ${LOGFILE} | wc -l`
-ftp_sl=`grep "OK LOGIN" ${LOGFILE} | wc -l`
-ftp_fl=`grep "FAIL LOGIN" ${LOGFILE} | wc -l`
-ftp_su=`grep "OK UPLOAD" ${LOGFILE} | wc -l`
-ftp_fu=`grep "FAIL UPLOAD" ${LOGFILE} | wc -l`
-ftp_sd=`grep "OK DOWNLOAD" ${LOGFILE} |wc -l`
-ftp_fd=`grep "FAIL DOWNLOAD" ${LOGFILE} | wc -l`
-ftp_sde=`grep "OK DELETE" ${LOGFILE} |wc -l`
-ftp_fde=`grep "FAIL DELETE" ${LOGFILE} | wc -l`
+if [ -f "${logfile}" ]; then
+	ftp_c=$(grep -c "CONNECT" "${logfile}")
+	ftp_sl=$(grep -c "OK LOGIN" "${logfile}")
+	ftp_fl=$(grep -c "FAIL LOGIN" "${logfile}")
+	ftp_su=$(grep -c "OK UPLOAD" "${logfile}")
+	ftp_fu=$(grep -c "FAIL UPLOAD" "${logfile}")
+	ftp_sd=$(grep -c "OK DOWNLOAD" "${logfile}")
+	ftp_fd=$(grep -c "FAIL DOWNLOAD" "${logfile}")
+	ftp_sr=$(grep -c "OK RENAME" "${logfile}")
+	ftp_fr=$(grep -c "FAIL RENAME" "${logfile}")
+	ftp_sde=$(grep -c "OK DELETE" "${logfile}")
+	ftp_fde=$(grep -c "FAIL DELETE" "${logfile}")
+fi
```

Comment:
Does grep -c not return "0"?

That awkward is definitely better I'll throw that in. I'd initially just modified existing and kept roughly the same method - but you are right to take this further while someone's attention is on it!